_source/_posts/2021-08-09-aws-toolkit-for-visual-studio.md

```diff
@@ -0,0 +1,415 @@
+                options.DefaultSignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                options.DefaultChallengeScheme = OktaDefaults.MvcAuthenticationScheme;
+            })
+           .AddCookie()
+           .AddOktaMvc(new OktaMvcOptions
+           {
+                // Replace these values with your Okta configuration
+               OktaDomain = Configuration.GetValue<string>("Okta:Domain"),
+               ClientId = Configuration.GetValue<string>("Okta:ClientId"),
+               ClientSecret = Configuration.GetValue<string>("Okta:ClientSecret"),
+               CallbackPath = Configuration.GetValue<string>("Okta:CallbackPath")
+           });
+
+            services.AddControllersWithViews();
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+            else
+            {
+                app.UseExceptionHandler("/Home/Error");
+                // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
+                app.UseHsts();
+            }
+            app.UseHttpsRedirection();
+            app.UseStaticFiles();
+
+            app.UseRouting();
+
+            app.UseAuthentication();
+            app.UseAuthorization();
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapControllerRoute(
+                    name: "default",
+                    pattern: "{controller=Home}/{action=Index}/{id?}");
+            });
+        }
+    }
+}
+```
+
+This file brings it all together. You have registered the Okta component using the `appsettings` file that matches your `ASPNETCORE_ENVIRONMENT` variable. Now you can easily change your Okta application depending on your environment.
+
+## Create your Okta Development Application
+
+You'll need to create an Okta application for your development environment. You can use the [Okta CLI tool](https://cli.okta.com/). Download the CLI tool and run `okta login`. This will walk you through the log in process. Once that is completed run `okta apps create` and follow the prompts on the screen. Select *web* as your application type. For your *Redirect URI*, use the web address provided from your application and append `/okta/callback` to it. For your *Post Logout RedirectURI*, use the same domain. This time, append `/signout/callback` to it. If you have multiple authorization servers you can select *default* for this application. After a moment, the CLI will write your Client Id, Client Secret, and Issuer to a file called `.okta.env` in the same directory you were working in. Take these values and add them to your `appsettings.Development.json` file.
```

Comment:
```suggestion
You'll need to create an Okta application for your development environment. You can use the [Okta CLI tool](https://cli.okta.com/). Download the CLI tool and run `okta login`. This will walk you through the login process. Once that is completed run `okta apps create` and follow the prompts on the screen. Select *web* as your application type. For your *Redirect URI*, use the web address provided from your application and append `/okta/callback` to it. For your *Post Logout RedirectURI*, use the same domain. This time, append `/signout/callback` to it. If you have multiple authorization servers you can select *default* for this application. After a moment, the CLI will write your Client Id, Client Secret, and Issuer to a file called `.okta.env` in the same directory you were working in. Take these values and add them to your `appsettings.Development.json` file.
```
the noun form: the action of logging in